components/brave_ads/browser/brave_ads_driver.cc

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/browser/brave_ads_driver.h"
+
+#include <utility>
+
+#include "brave/components/brave_ads/browser/ads_service.h"
+#include "brave/components/brave_ads/browser/brave_ads_driver_delegate.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+
+namespace brave_ads {
+
+BraveAdsDriver::BraveAdsDriver(AdsService* ads_service,
+                               brave_rewards::RewardsService* rewards_service,
+                               std::unique_ptr<BraveAdsDriverDelegate> delegate)
+    : ads_service_(ads_service),
+      rewards_service_(rewards_service),
+      delegate_(std::move(delegate)) {
+  DCHECK(delegate_);
+}
+
+BraveAdsDriver::~BraveAdsDriver() {}
+
+void BraveAdsDriver::RequestAdsEnabled(RequestAdsEnabledCallback callback) {
+  if (!ads_service_ || !rewards_service_) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  const bool is_ads_enabled = ads_service_->IsEnabled();
```

Comment:
do not use these methods, they should be deprecated. Use the prefs instead

Reply:
@aseren as above these will be deprecated once we fix https://github.com/brave/brave-browser/issues/17500

Reply:
Thank you! 
Seems like `RewardsServiceImpl::IsRewardsEnabled` call is redundant. I will remove it.

Reply:
Done